esx_service/cli/vmdkops_admin.py

```diff
@@ -563,11 +563,14 @@ def get_creation_info(metadata):
 
 
 def get_attached_to(metadata):
-    """ Return which VM a volume is attached to based on its metadata """
+    """ Return which VM a volume is attached to based on its metadata. """
     try:
         vm_name = vmdk_ops.vm_uuid2name(metadata[kv.ATTACHED_VM_UUID])
         if not vm_name:
+            if metadata[kv.ATTACHED_VM_UUID]:
+                return metadata[kv.ATTACHED_VM_UUID]
```

Comment:
i think logging bitgh vmname and uuid and resetting vmname to "" if it's not avail is a way to go since it adds predictibility.

On ESX name - I am not sure which ESX ? where it's attached ? Where it was created ?